src/v/storage/segment_utils.cc

```diff
@@ -550,6 +550,7 @@ ss::future<std::optional<size_t>> do_self_compact_segment(
 }
 
 ss::future<> build_compaction_index(
+  ss::lw_shared_ptr<storage::segment> s,
```

Comment:
i think it will be enough to pass segment dirty offset into this method, the segment itself isn't required